sdks/python/apache_beam/examples/complete/game/game_stats.py

```diff
@@ -0,0 +1,285 @@
+    # Get the sum of scores for each user.
+    sum_scores = (
+        user_scores
+        | 'SumUsersScores' >> beam.CombinePerKey(sum)
+    )
+
+    # Extract the score from each element, and use it to find the global mean.
+    global_mean_score = (
+        sum_scores
+        | beam.Values()
+        # NOTE: DataflowRunner currently does not support combiners
+        | beam.CombineGlobally(combiners.MeanCombineFn()).as_singleton_view()
+    )
+
+    # Filter the user sums using the global mean.
+    filtered = (
+        sum_scores
+        # Use the derived mean total score (global_mean_score) as a side input.
+        | 'ProcessAndFilter' >> beam.Filter(
+            lambda (_, score), global_mean:\
+                score > global_mean * self.SCORE_WEIGHT,
+            global_mean_score)
+    )
+    return filtered
+
+
+class UserSessionActivity(beam.DoFn):
+  """Calculate and output an element's session duration, in seconds."""
+  def process(self, elem, window=beam.DoFn.WindowParam):
+    yield (window.end.micros - window.start.micros) / 1000000
+
+
+def run(argv=None):
+  """Main entry point; defines and runs the hourly_team_score pipeline."""
+  parser = argparse.ArgumentParser()
+
+  parser.add_argument('--project',
+                      type=str,
+                      required=True,
+                      help='GCP Project ID for the output BigQuery Dataset.')
+  parser.add_argument('--topic',
+                      type=str,
+                      required=True,
+                      help='Pub/Sub topic to read from')
+  parser.add_argument('--dataset',
+                      type=str,
+                      required=True,
+                      help='BigQuery Dataset to write tables to. '
+                      'Must already exist.')
+  parser.add_argument('--table_name',
+                      type=str,
+                      default='game_stats',
+                      help='The BigQuery table name. Should not already exist.')
+  parser.add_argument('--fixed_window_duration',
+                      type=int,
+                      default=60,
+                      help='Numeric value of fixed window duration for user '
+                           'analysis, in minutes')
+  parser.add_argument('--session_gap',
+                      type=int,
+                      default=5,
+                      help='Numeric value of gap between user sessions, '
+                           'in minutes')
+  parser.add_argument('--user_activity_window_duration',
+                      type=int,
+                      default=30,
+                      help='Numeric value of fixed window for finding mean of '
+                           'user session duration, in minutes')
+
+  args, pipeline_args = parser.parse_known_args(argv)
+
+  # We use the save_main_session option because one or more DoFn's in this
+  # workflow rely on global context (e.g., a module imported at module level).
+  pipeline_args += ['--save_main_session']
```

Comment:
<!--thread_id:cc_128807623_t; commit:95d457e2ad40bb613cb8abda68903ad026f0cccd; resolved:0-->
<!--section:context-quote-->
> **davidcavazos** wrote:
> OK, so I made a note that this argument is not yet supported in DataflowRunner, I also added a note to the header that this example is not yet runnable in DataflowRUnner.
> 
> I'm also marking the ``leader_board.py`` example as not runnable yet since it also uses ``--save_main_session`` with streaming.

<!--section:body-->
Do you want to keep this flag here, if it is not going to run on `DataflowRunner` for now?